src/main/java/org/springframework/data/couchbase/repository/support/CrudMethodMetadataPostProcessor.java

```diff
@@ -161,10 +164,10 @@ public Object invoke(MethodInvocation invocation) throws Throwable {
 				try {
 					return invocation.proceed();
 				} finally {
-					TransactionSynchronizationManager.unbindResource(method);
+					// TransactionSynchronizationManager.unbindResource(method);
 				}
 			} finally {
-				currentInvocation.set(oldInvocation);
+				// currentInvocation.set(oldInvocation);
```

Comment:
unbinding the method and resetting the currentInvocation is problematic for  reactive calls - where the bound metadata maybe accessed in the Reactive Flux *after* the invocation is complete. 
Furthermore - unbinding/resetting is not necessary as this all occurs in the same thread.